### ISSUE TITLE
Add JudgeLM as a way to evaluate and compare historical test runs

### ISSUE THREAD

**Issue by penguine-ip** [confident-ai/deepeval#262]
Currently, metrics are computed based on test cases that run during evaluation. However, there's currently no way to compare historical test runs' performances except for comparing metric scores. Whilst this is a good way, another way to choose the best test run would be to implement a JudgeLM (eg. https://github.com/baaivision/JudgeLM) to choose the best test run out of a set of test runs.

```
from deepeval import judge

best_performing_test_run = None
for test_run in test_runs:
      best_performing_test_run = judge(test_run, best_performing_test_run)
```

This assumes the test runs were ran using the same test cases.

**Comment by Anindyadeep** [comment]
Do you want to an LLM to judge another LLM here? Like can you more elaborate the issue. May be I need to go through the paper to understand this more. But yes, this can be very useful when pre-training or fine-tuning models and when we are in the middle of Training.

**Comment by penguine-ip** [comment]
So JudgeLM will be used to pick the best historical test run out of a set of a test runs, but it would be difficult for this to be a feature purely for the package

**Comment by Anindyadeep** [comment]
Yes, that could be.

**Comment by AyushChauhan910** [comment]
Hi @penguine-ip , I'd like to take this on if the scope below works.

I think the "purely a package feature" concern comes from treating "historical" as something DeepEval has to fetch and store. I'd scope this without that: a judge() utility that takes two (or more) already-in-memory TestRun objects (or lists of TestResult) that the user already has from their own evaluate() calls or loaded from disk, plus a judge model. No storage or backend involved, so it stays local like the rest of the SDK.

Rough shape:

from deepeval import judge

result = judge(
    test_runs=[run_a, run_b],
    model=judge_model,
)

result.winner -> the better TestRun
result.reasoning -> why

Internally this pairs matching test cases across runs by input/dataset alignment, has the judge model compare outputs pairwise per case, and aggregates to a run-level winner with per-case reasoning attached. I'd add error handling for runs that weren't evaluated on the same test cases, per your original note.

I've built something in this shape before, an LLM-as-judge layer with inter-rater calibration for scoring agent outputs in production, so I'm comfortable owning the judge-comparison logic and the tests.

Does this address the scoping concern, or did you have something more backend-dependent in mind?
